config/env/transform.go

```diff
@@ -1,9 +1,23 @@
 // Package configenv implements a TransformationProvider that replaces objects on
 // the form: {$env: "VAR"} with the value of the environment variable VAR.
+//
+// An additional key 'type' may be given as {$env: "VAR", type: "TYPE"}, which
+// determines how the environment variable value should be parsed. By default
+// the TYPE is 'string', but the following types are supported:
+//   - string, value is return as string (default),
+//   - number, value is parsed a number,
```

Comment:
No, the object hierarchy uses types matching what you get after `var t interface{}; json.Unmarshal(..., &t)`

So all integers will be loaded in as `float64`.. Consider these config transformers as functions that operate on JSON equivalent data structures:
 * `map[string]interface{}`
 * `interface{}`
 * `bool`
 * `float64`
 * `string`
 * `nil`

Maybe I should make some checks for that...